cmd/gateway/setup_test.go

```diff
@@ -263,4 +263,25 @@ var _ = Describe("Main", func() {
 			}) // should fail with invalid name
 		}) // gatewayclass validation
 	}) // CLI argument validation
+
+	Describe("environment variable validaton", func() {
+		type testCase struct {
+			expSubMsg string
+			podIP     string
+			expErr    bool
+		}
+		DescribeTable("should validate the POD_IP env var",
```

Comment:
I see a mismatch here. We don't really validate the POD_IP env variable. We just validate any IP, because the function `ValidatePodIP` takes a string argument and is not aware of env variables.

I wonder if we can simply things and unexport `ValidatePodIP` and have a simple unit test for it, rather than try to fit it into more complex tests for validating cli argument.

@kate-osborn thoghts?

Reply:
I can fix up the naming, though if we want to unit test it, it has to be exported because our tests live in a different package `main_test`.

Reply:
I could do that, if we're okay with `main_test.go` to be a part of the `main` package, instead of the `main_test` package, as `setup_test.go` is.

Reply:
this is fine with me 

@kate-osborn ?